resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh

```diff
@@ -60,14 +60,44 @@ if ! [ -z ${HADOOP_CONF_DIR+x} ]; then
   SPARK_CLASSPATH="$HADOOP_CONF_DIR:$SPARK_CLASSPATH";
 fi
 
+IGNORE_DEFAULT_DRIVER_JVM_OPTIONS=${IGNORE_DEFAULT_DRIVER_JVM_OPTIONS:-false}
```

Comment:
Can we have `DEFAULT_DRIVER_JVM_OPTIONS` instead of this flag?
1. By default, `DEFAULT_DRIVER_JVM_OPTIONS=-XX:OnOutOfMemoryError="kill -9 %p"` and it will be appended **before** `spark.driver.extraJavaOptions`.
2. If users unset `DEFAULT_DRIVER_JVM_OPTIONS`, then only `spark.driver.extraJavaOptions` works.

At (1), if `spark.driver.extraJavaOptions` has `-XX:OnOutOfMemoryError=`, it will supersede `DEFAULT_DRIVER_JVM_OPTIONS` because the last one is used.

Reply:
How can they unset the flag at runtime?

Reply:
@dongjoon-hyun the only option I see is setting `DEFAULT_DRIVER_JVM_OPTIONS` to empty string to signal the script to use the default, which is similar to setting that flag to true.

> If users unset DEFAULT_DRIVER_JVM_OPTIONS
How should the user do that?

Reply:
@dongjoon-hyun gentle ping :)

Reply:
Oops. Sorry for being late, @skonto . I'll take a look this PR tomorrow again~

Reply:
Initially, what I thought was something like the following
```
DEFAULT_DRIVER_JVM_OPTIONS=${DEFAULT_DRIVER_JVM_OPTIONS:--XX:OnOutOfMemoryError="kill -9 %p"}
```
And users do `export DEFAULT_DRIVER_JVM_OPTIONS=' '` for unset.

Reply:
@skonto . What do you think about the above? For me, the above looks more direct.

Reply:
Ok so that is what I concluded. Ok I can do that, it looks fine to me, the only issue is that this way we open the door to adding arbitrary options there. Initially that was what I didnt want to support, thus the flag. But we could be flexible here.